contracts/contracts/chains/ethereum/EthereumMessengers.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.12;
+
+import "OpenZeppelin/openzeppelin-contracts@4.8.0/contracts/access/Ownable.sol";
+
+import "../../../interfaces/IMessenger.sol";
+import "../../RestrictedCalls.sol";
+import "../../Resolver.sol";
+
+contract EthereumL1Messenger is IMessenger, RestrictedCalls {
+    function callAllowed(address, address) external view returns (bool) {
+        // The call must be delivered by the Resolver
+        // Since we are on the same chain as the Resolver
+        // we should never see an execution of this function.
+        revert("Unexpected state.");
+    }
+
+    // Simply forward the message to the target on L1
+    /* solhint-disable avoid-tx-origin */
+    function sendMessage(address target, bytes calldata message)
+        external
+        restricted(block.chainid)
+    {
+        target.call(message);
+    }
+}
+
+// In order to keep naming consistent between our rollup specific contracts
+// we are using `EthereumL2Messenger` here as a name even if this contract
+// is supposed to be deployed on L1.
+contract EthereumL2Messenger is IMessenger, Ownable, RestrictedCalls {
+    enum MessageStatus {
+        UNDEFINED,
+        DEFINED,
+        RELAYED
+    }
+
+    Resolver public immutable resolver;
+
+    /// Maps message hashes with their status.
+    mapping(bytes32 => MessageStatus) public messageHashes;
+
+    constructor(address resolver_) {
```

Comment:
why dont we use _resolver instead of resolver_ here ?

Reply:
No idea, saw this pattern being used in `ArbitrumL1Messenger`'s constructor.

@istankovic ?

Reply:
`resolver_` is correct style here, per https://docs.soliditylang.org/en/v0.8.19/style-guide.html#avoiding-naming-collisions. So this is fine.

Reply:
ok.